Example/PaymentSheet Example/PaymentSheet Example/ExampleCustomDeferredCheckoutViewController.swift

```diff
@@ -0,0 +1,332 @@
+        updateUI()
+    }
+
+    // MARK: - Helper methods
+
+    private func updateUI() {
+        // Disable buy and payment method buttons
+        buyButton.isEnabled = false
+        paymentMethodButton.isEnabled = false
+
+        fetchTotals { [weak self] in
+            guard let self = self else { return }
+            self.updateLabels()
+
+            // Update PaymentSheet with the latest `intentConfig`
+            self.paymentSheetFlowController.update(intentConfiguration: self.intentConfig) { error  in
+                if error != nil {
+                    // Retry
+                    self.updateUI()
+                } else {
+                    // Re-enable your "Buy" and "Payment method" buttons
+                    self.updateButtons()
+                  }
+            }
+        }
+    }
+
+    func updateButtons() {
+        paymentMethodButton.isEnabled = true
+
+        // MARK: Update the payment method and buy buttons
+        if let paymentOption = paymentSheetFlowController.paymentOption {
+            paymentMethodButton.setTitle(paymentOption.label, for: .normal)
+            paymentMethodButton.setTitleColor(.black, for: .normal)
+            paymentMethodImage.image = paymentOption.image
+            buyButton.isEnabled = true
+        } else {
+            paymentMethodButton.setTitle("Select", for: .normal)
+            paymentMethodButton.setTitleColor(.systemBlue, for: .normal)
+            paymentMethodImage.image = nil
+            buyButton.isEnabled = false
+        }
+    }
+
+    func updateLabels() {
+        hotDogQuantityLabel.text = "\(Int(hotDogStepper.value))"
+        saladQuantityLabel.text = "\(Int(saladStepper.value))"
+
+        subtotalLabel.text = "\(currencyFormatter.string(from: NSNumber(value: computedTotals.subtotal)) ?? "")"
+        salesTaxLabel.text = "\(currencyFormatter.string(from: NSNumber(value: computedTotals.tax)) ?? "")"
+        totalLabel.text = "\(currencyFormatter.string(from: NSNumber(value: computedTotals.total)) ?? "")"
+    }
+
+    func displayAlert(_ message: String) {
+        let alertController = UIAlertController(title: "", message: message, preferredStyle: .alert)
+        let OKAction = UIAlertAction(title: "OK", style: .default) { (_) in
+            alertController.dismiss(animated: true) {
+                self.navigationController?.popViewController(animated: true)
+            }
+        }
+        alertController.addAction(OKAction)
+        present(alertController, animated: true, completion: nil)
+    }
+
+    // MARK: Server-side confirm handler
+
+    func serverSideConfirmHandler(_ paymentMethodID: String,
+                                  _ shouldSavePaymentMethod: Bool,
+                                  _ intentCreationCallback: @escaping (Result<String, Error>) -> Void) {
+        // Create an intent on your server and invoke `intentCreationCallback` with the client secret
+        createIntent(paymentMethodID: paymentMethodID) { result in
```

Comment:
We need to pass `shouldSavePaymentMethod` to the server too and have it set `setup_future_usage = off_session` if it's `true`.  I've realized this isn't in the guide either, I'll add it there.

Reply:
how does `shouldSavePaymentMethod` relate with the subscribe switch? Should we send the `setup_future_usage = off_session` if switch is on or if `shouldSavePaymentMethod` is true?

Reply:
Yep that sounds right.